pkg/informers/kube.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package informers
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+)
+
+func NewKubeInformerFactories(
```

Comment:
I think it's a little bit confusing that this has `Kube` in the name. I kind of think that all of this code is `kube`-related, since all of it is dealing with informers. Why not simply `NewInformerFactories`?

Reply:
hmm I see that this is because of the fact that this func takes a kubernetes.Interface as opposed to the monitoring client ://// I wish we could abstract this better to be less repetitive. If only we had generics :))

Reply:
yes this is unfortunate indeed :-/

I tried to factor as much as possible into the `ForResource` informer, but some duplication stayed. Generics would've helped (or code generation).